CarpetX/src/io_openpmd.cxx

```diff
@@ -1771,36 +1827,74 @@ void carpetx_openpmd_t::OutputOpenPMD(const cGH *const cctkGH,
         assert(cactus_dk > 0);
         assert(cactus_np > 0);
         for (int vi = 0; vi < numvars; ++vi) {
-          const CCTK_REAL *const var_ptr =
-              groupdata.data.at(tl).data() + vi * cactus_np;
+          const void *const var_ptr =
+              groupdata.data.at(tl).data_at(vi * cactus_np);
           if (output_ghosts || intbox == extbox) {
-            const CCTK_REAL *const ptr = var_ptr;
-#if OPENPMDAPI_VERSION_GE(0, 15, 0)
-            record_components.at(vi).storeChunkRaw(ptr, start, count);
-#else
-            record_components.at(vi).storeChunk(openPMD::shareRaw(ptr), start,
-                                                count);
+#if !OPENPMDAPI_VERSION_GE(0, 15, 0)
+#define storeChunkRaw(ptr, start, count)                                       \
+  storeChunk(openPMD::shareRaw(ptr), start, count)
 #endif
+            switch (cgroup.vartype) {
+            case CCTK_VARIABLE_REAL:
+              record_components.at(vi).storeChunkRaw(
+                  static_cast<CCTK_REAL const *const>(var_ptr), start, count);
```

Comment:
This should be `static_cast<CCTK_REAL const *>`, the additional `const` doesn't have any effect since the pointer is an rvalue.

Reply:
Done in 016fec7